src/qibolab/transpilers/general_connectivity.py

```diff
@@ -0,0 +1,409 @@
+        elif qubits == nodes:
+            new_circuit = Circuit(nodes)
+        else:
+            log.warning(
+                "You are using more physical qubits than required by the circuit, some qubits will be added to the circuit"
+            )
+            new_circuit = Circuit(nodes)
+        self.transpiled_circuit = new_circuit
+
+    def init_mapping_circuit(self, circuit, qubit_map):
+        """Initial qubit mapping of the transpiled qibo circuit
+
+        Args:
+            circuit (qibo.Circuit): transpiled qibo circuit.
+            qubit_map (np.array): initial qubit mapping.
+
+        Returns:
+            new_circuit (qibo.Circuit): transpiled circuit mapped with initial qubit mapping.
+        """
+        new_circuit = Circuit(self._connectivity.number_of_nodes())
+        for gate in circuit.queue:
+            new_circuit.add(gate.on_qubits({q: qubit_map[q] for q in gate.qubits}))
+        return new_circuit
+
+    def add_gates(self, qibo_circuit, matched_gates):
+        """Add one and two qubit gates to transpiled circuit until connectivity is matched
+
+        Args:
+            qibo_circuit (qibo.Circuit): circuit to be transpiled.
+            matched_gates (int): number of two qubit gates that can be applied with the current qubit mapping
+        """
+        index = 0
+        while self._circuit_position < len(qibo_circuit.queue):
+            gate = qibo_circuit.queue[self._circuit_position]
+            if len(gate.qubits) == 1:
+                self.transpiled_circuit.add(gate.on_qubits({gate.qubits[0]: self._qubit_map[gate.qubits[0]]}))
+                self._circuit_position += 1
+            else:
+                index += 1
+                if index == matched_gates + 1:
+                    break
+                else:
+                    self.transpiled_circuit.add(
+                        gate.on_qubits(
+                            {
+                                gate.qubits[0]: self._qubit_map[gate.qubits[0]],
+                                gate.qubits[1]: self._qubit_map[gate.qubits[1]],
+                            }
+                        )
+                    )
+                    self._circuit_position += 1
+
+    def add_swaps(self, path, meeting_point):
+        """Add swaps to the transpiled circuit to move qubits
+
+        Args:
+            path (list): path to move qubits.
+            meeting_point (int): qubit meeting point in the path.
+        """
+        forward = path[0 : meeting_point + 1]
+        backward = path[meeting_point + 1 :]
+        if len(forward) > 1:
+            for i in range(len(forward) - 1):
+                self.transpiled_circuit.add(gates.SWAP(self._qubit_map[forward[i]], self._qubit_map[forward[i + 1]]))
+        if len(backward) > 1:
+            for i in range(len(backward) - 1, 0, -1):
+                self.transpiled_circuit.add(gates.SWAP(self._qubit_map[backward[i]], self._qubit_map[backward[i - 1]]))
```

Comment:
Same comment of above, but here you need to skip the last step, not the first (of course if you have a suitable iterator you don't have to implement it manually)